R/modify.R

```diff
@@ -22,44 +22,62 @@
 json_modify_text <- function(text, json_path, value, spaces = 4) {
   stopifnot(is.character(text))
   text <- paste(text, collapse = '\n')
-  if (is.null(value)) value <- V8::JS('undefined')
+  if (is.null(value)) {
+    value <- V8::JS('undefined')
+  }
   opts <- if (length(spaces)) {
     list(formattingOptions = list(insertSpaces = spaces > 0, tabSize = spaces))
   }
+  if (!is.null(json_parse_errors(text))) {
+    stop("Can't modify when there are existing parse errors.")
+  }
   jsonc$call('json_modify', text, as.list(json_path), value, opts)
 }
 
 #' @export
 #' @rdname jsonedit
 #' @param file path to file on disk. Will be created if it does not exist.
 json_modify_file <- function(file, json_path, value, spaces = 4) {
-  json <- if (file.exists(file)) {
+  text <- if (file.exists(file)) {
     rawToChar(readBin(file, raw(), file.info(file)$size))
   } else {
     "{}"
   }
-  out <- json_modify_text(json, json_path, value, spaces)
+  out <- json_modify_text(text, json_path, value, spaces)
   writeLines(out, file)
 }
 
 #' @export
 #' @rdname jsonedit
 #' @param spaces number of spaces to indent. Use 0 for tabs.
 json_format_text <- function(text, spaces = 4) {
-  stopifnot(is.character(json))
-  json <- paste(json, collapse = '\n')
+  stopifnot(is.character(text))
+  text <- paste(text, collapse = '\n')
   opts <- list(insertSpaces = spaces > 0, tabSize = spaces)
-  jsonc$call('json_format', json, opts)
+  if (!is.null(json_parse_errors(text))) {
+    stop("Can't format when there are existing parse errors.")
+  }
+  jsonc$call('json_format', text, opts)
 }
```

Comment:
`json` wasn't actually defined here so this previously failed. i think you meant to use `text`, and ive standardized all functions to just use `text` now